src/main/common/vector.h

```diff
@@ -26,124 +26,44 @@
 
 #include "common/maths.h"
 
-typedef union {
+typedef union vector2_u {
     float v[2];
     struct {
-       float x,y;
+       float x, y;
     };
-} fpVector2_t;
+} vector2_t;
 
-
-typedef union {
+typedef union vector3_u {
     float v[3];
     struct {
        float x, y, z;
     };
-} fpVector3_t;
+} vector3_t;
 
-typedef struct {
+typedef struct matrix33_s {
     float m[3][3];
-} fpMat33_t;
-
-static inline fpVector3_t * vectorZero(fpVector3_t *v)
```

Comment:
I'm not sure if LTO is smart enough to handle all inlining (math functions may end up in different compilation unit).
Inline functions are very aggressively optimized, removing all temporaries, propagating constant expressions (all multiplications with zero are avoided etc.
And gcc  should ignore `inline` if it decides that it is not worth it, so may be safe choice

Reply:
My thought was: We can probably save some flash in the future by not allowing inlining. Right now it does not make a huge difference because we don't use the functions often but that might be subject to change.

Reply:
IMU code assumes that operations will be inlined and all zero multiplications factored out. Without it, it will be considerably slower. 
As extreme case, with inlining, it is possible to extract row/column of orientation matrix using matrix orientation by [0,0,1] - get up vector in earth frame. It is easy to read and it is optimized out.

Reply:
@ledvinap Alright, I will now try to finish up this PR. This means I need to get rid of vector.c again and inline all of the definitions in the header file (revert back to what you did), right?

Reply:
I'd put very short functions into .h (*Zero, scale, equal) , longer ones (matrix/vector multiplication, matrix building) into .c. Not sure about cross, but probably to .c

Reply:
I'm asking because everything that is `static inline` needs to be in the header file. So, if matrix multiplication (for example) is supposed to be optimized, I'd need to pull it out of the .c and put it in the .h, no matter how big the function is.

Basically I'm just asking what to do exactly to make sure the IMU code is as fast as possible. Right now to me it looks like I need to put all functions into `vector.h` to make sure they can be statically inlined.

Reply:
There is almost no difference in generated code. Register allocation (and saving) is slightly different, but number of instructions seems to be the same (non-inlined version is 1 assembly listing line longer out of 15k lines ).

Reply:
Oh wow, thanks for testing. In this case I'm leaving it as is for now (not explicitly inlined). Feel free to change whatever you want after this PR is merged.